buildSrc/src/main/groovy/BuildkitBuild.groovy

```diff
@@ -0,0 +1,399 @@
+import org.gradle.api.provider.ListProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.Optional
+import org.gradle.api.provider.ProviderFactory
+import org.gradle.api.Task
+import static org.gradle.api.tasks.PathSensitivity.RELATIVE
+import javax.inject.Inject
+import java.net.Socket
+import java.text.SimpleDateFormat
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardCopyOption
+
+/**
+ *  Task to publish worker images to artifactory using buildkit
+ *  https://r3-cev.atlassian.net/wiki/spaces/CB/pages/4063035406/BuildKit
+ */
+abstract class BuildkitBuild extends Exec {
+
+    private final String projectName = project.name
+    private final String version = project.version
+    private def gitTask
+    private def gitLogTask
+    private def releaseType
+
+    @Inject
+    protected abstract ObjectFactory getObjects()
+
+    @Inject
+    protected abstract ProviderFactory getProviderFactory()
+
+    // Credentials used for authentication to docker and artifactory
+    @Input
+    @Optional
+    final Property<String> registryUsername = getObjects().property(String).
+            convention(getProviderFactory().environmentVariable("CORDA_ARTIFACTORY_USERNAME")
+                    .orElse(getProviderFactory().gradleProperty("cordaArtifactoryUsername"))
+                    .orElse(getProviderFactory().systemProperty("corda.artifactory.username"))
+            )
+
+    @Input
+    @Optional
+    final Property<String> registryPassword = getObjects().property(String).
+            convention(getProviderFactory().environmentVariable("CORDA_ARTIFACTORY_PASSWORD")
+                    .orElse(getProviderFactory().gradleProperty("cordaArtifactoryPassword"))
+                    .orElse(getProviderFactory().systemProperty("corda.artifactory.password"))
+            )
+
+    // Property to set images for release 
+    @Input
+    final Property<Boolean> releaseCandidate = getObjects().property(Boolean).convention(false)
+
+    //Property for nightly builds
+    @Input
+    final Property<Boolean> nightlyBuild = getObjects().property(Boolean).convention(false)
+
+    // Property for pre-test
+    @Input
+    final Property<Boolean> preTest = getObjects().property(Boolean).convention(false)
+
+    // Property to set the build tool used to create an image
+    // Currently supported tools are docker buildx and native builkit 
+    // NOTE: native buidkit requires port forwarding to the aws buildkit cluster
+    @Input
+    final Property<Boolean> isBuildx = getObjects().property(Boolean).convention(true)
+
+    // Property for loading images into docker
+    @Input
+    final Property<Boolean> useDocker = getObjects().property(Boolean).convention(false)
```

Comment:
Can we rename this to useDockerDaemon please. as technically we are using docker in some capacity either way.

And default it to true ( And pipeline can override value then)